GH-11085 Migrate Team.GetAllTeamPageListing to Sync by default

lurcio opened the pull request:
#### Summary
Migrate "Team.GetAllTeamPageListing" to Sync by default

#### Ticket Link
Fixes https://github.com/mattermost/mattermost-server/issues/11085

mounicapaladugu commented on store/sqlstore/team_store.go:
```diff
@@ -450,22 +450,19 @@ func (s SqlTeamStore) GetAllTeamListing() store.StoreChannel {
-		}
+	var data []*model.Team
+	if _, err := s.GetReplica().Select(&data, query, map[string]interface{}{"Offset": offset, "Limit": limit}); err != nil {
+		return nil, model.NewAppError("SqlTeamStore.GetAllTeamListing", "store.sql_team.get_all_team_listing.app_error", nil, err.Error(), http.StatusInternalServerError)
```
Can you change this error message to GetAllTeamPageListing and run make i18n-extract to extract the strings from the source code to the file i18n/en.json? Also modify store.sql_team.get_all_team_listing.app_error --> store.sql_team.get_all_team_page_listing.app_error

lurcio replied:
@mounicapaladugu Sure, will do. I'll take a look tomorrow, it's late here 😄

jespino commented on app/team.go:
```diff
@@ -649,11 +649,11 @@ func (a *App) GetAllPublicTeams() ([]*model.Team, *model.AppError) {
 }
 
 func (a *App) GetAllPublicTeamsPage(offset int, limit int) ([]*model.Team, *model.AppError) {
-	result := <-a.Srv.Store.Team().GetAllTeamPageListing(offset, limit)
```
You can just `return a.Srv.Store.Team().GetAllTeamPageListing(offset, limit)`

jespino commented on store/sqlstore/team_store.go:
```diff
@@ -450,22 +450,19 @@ func (s SqlTeamStore) GetAllTeamListing() store.StoreChannel {
-			result.Err = model.NewAppError("SqlTeamStore.GetAllTeamListing", "store.sql_team.get_all_team_listing.app_error", nil, err.Error(), http.StatusInternalServerError)
-			return
-		}
+	var data []*model.Team
```
nit: Probably we can rename here `data` to `teams` which is more meaningful.

reflog commented on store/storetest/team_store.go:
```diff
@@ -536,20 +536,17 @@ func testGetAllTeamPageListing(t *testing.T, ss store.Store) {
-	} else {
-		teams := r1.Data.([]*model.Team)
+	teams, err := ss.Team().GetAllTeamPageListing(0, 10)
+	require.Nil(t, err)
```
Nit: `require.NoError` is more appropriate here. 0/5

jespino replied:
No is not, it generates certain problems related to we don't return an error, we return a model.AppError, which is not exactly the same.

reflog replied:
Ouch. Thanks for clarifying!

reflog commented on store/storetest/team_store.go:
```diff
@@ -561,36 +558,30 @@ func testGetAllTeamPageListing(t *testing.T, ss store.Store) {
-	} else {
-		teams := r1.Data.([]*model.Team)
+	teams, err = ss.Team().GetAllTeamPageListing(0, 4)
+	require.Nil(t, err)
```
Same nitpick as above

lurcio commented:
@mounicapaladugu sorry for the delay in looking at this. I have incorporated your feedback. Running `make i18n-extract` has added more changes than I anticipated and I think this is breaking the build at the `Check i18n` step. Is this OK? Could you take a look please?

@jespino Thank you for your feedback, I have changed my PR as per your review.

mounicapaladugu commented:
Hi @lurcio. Don't worry about the modification if it adds too much overhead. You can revert the error message to whatever it was before. Thanks!!
